### PR TITLE
fix: run jsdom only if needed html2x (fixes #258)

### LINKED ISSUE #258: Document is parsed twice with JSDOM
**Is your feature request related to a problem? Please describe.**
For crosswalk we use the html2md in a serverless function. That works fine. However it does not perform very well yet. JSDOM adds about 300ms for the document I am testing.

Having a look at a performance measurement it seems that the importer is running JSDOM twice on the html. 

(1) https://github.com/adobe/helix-importer/blob/main/src/importer/HTML2x.js#L239
(2) https://github.com/adobe/helix-importer/blob/main/src/importer/PageImporter.js#L300

My assumption would be, that if I provide a document object (not a string) to html2md, it is used as is and no further serialisation and reparsing happens. That would also allow us to use different dom parsers depending on the use case or context, e.g.

- not the full feature set of JSDOM is needed we may want to use https://github.com/WebReflection/linkedom (which in my test is significantly faster than using JSDOM)
- in browser we may want to use the native DOMParser

**Describe the solution you'd like**
- ideally we could remove the double parsing using JSDOM

**Describe alternatives you've considered**
- none yet

### PR DESCRIPTION
**Pull request by Buuhuu**
Please ensure your pull request adheres to the following guidelines:
- [x] make sure to link the related issues in this description
- [x] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues

Fixes #258

Thanks for contributing!

### CODE REVIEW

**Comment by kptdobe** [comment]
Probably a valid feedback. My main issue is that I am working on a major redesign which includes removing JSDOM (see https://github.com/adobe/helix-importer/pull/260). But your fix is probably still valid. Will look at it in the context of the redesign.

File: src/importer/PageImporter.js [hunk 1]
```diff
@@ -194,7 +194,7 @@ export default class PageImporter {
   }
 
   cleanup(document) {
-    DOMUtils.remove(document, ['script', 'hr']);
+    DOMUtils.remove(document, ['noscript', 'script', 'hr']);
```
Comment (kptdobe) [inline src/importer/PageImporter.js:197]: This is out of the scope for this PR...

**Review by kptdobe — CHANGES_REQUESTED** [review]
The `noscript` cleanup is out of scope and should be removed from this PR. The rest is great, thanks @Buuhuu. I have already ported your feedback in the upcoming `no-jsdom` branch:
- https://github.com/adobe/helix-importer/pull/260/commits/12be7bfd0503f5a4a2545e7bce74b2327180faa6
- https://github.com/adobe/helix-importer/pull/260/commits/ddfd85ca4ddb31dc27ad94bcea68c807a61f2568

File: src/importer/PageImporter.js (hunk 1 above)
Comment (Buuhuu) [reply src/importer/PageImporter.js:197]: If provided a html string, than the noscripts are removed, see

https://github.com/adobe/helix-importer/pull/259/files/07a8a5266dbb7116fa88f370f930077a19a44977#diff-b966904efa617d33768f65ed8352810264c7991e6758a9bf9ce41142cb6e8516L300

I thought it makes sense to have the same behaviour if a document object is provided.

**Comment by kptdobe** [comment]
Closing this PR as it has been integrated with https://github.com/adobe/helix-importer/pull/260 which has now been released. 
@Buuhuu you might want to check the latest importer release which in your case might require some minor adjustments if you use string representation (like providing `createDocumentFromString` function - see example in test: https://github.com/adobe/helix-importer/blob/main/test/importers/HTML2x.spec.js#L56)

**Pull request closed without merging**